lib/rules/no-restricted-properties.js

```diff
@@ -142,40 +142,25 @@ module.exports = {
             }
         }
 
-        /**
-         * Checks property accesses in a destructuring assignment expression, e.g. `var foo; ({foo} = bar);`
-         * @param {ASTNode} node An AssignmentExpression or AssignmentPattern node
-         * @returns {undefined}
-         */
-        function checkDestructuringAssignment(node) {
-            if (node.right.type === "Identifier") {
-                const objectName = node.right.name;
-
-                if (node.left.type === "ObjectPattern") {
-                    node.left.properties.forEach(property => {
-                        checkPropertyAccess(node.left, objectName, astUtils.getStaticPropertyName(property));
-                    });
-                }
-            }
-        }
-
         return {
             MemberExpression(node) {
                 checkPropertyAccess(node, node.object && node.object.name, astUtils.getStaticPropertyName(node));
             },
-            VariableDeclarator(node) {
-                if (node.init && node.init.type === "Identifier") {
-                    const objectName = node.init.name;
-
-                    if (node.id.type === "ObjectPattern") {
-                        node.id.properties.forEach(property => {
-                            checkPropertyAccess(node.id, objectName, astUtils.getStaticPropertyName(property));
-                        });
-                    }
+            ObjectPattern(node) {
+                let objectName;
+
+                if (node.parent.type === "VariableDeclarator") {
+                    objectName = node.parent.init && node.parent.init.type === "Identifier" ? node.parent.init.name : "";
+                } else if ((node.parent.type === "AssignmentExpression" || node.parent.type === "AssignmentPattern") && node.parent.right.type === "Identifier") {
+                    objectName = node.parent.right.name;
+                } else {
+                    objectName = "";
                 }
```

Comment:
Can we set `null` as the indicator that the object name is unknown?

```suggestion
                let objectName = null;

                if (node.parent.type === "VariableDeclarator") {
                    if (node.parent.init && node.parent.init.type === "Identifier") {
                        objectName = node.parent.init.name;
                    }
                } else if (node.parent.type === "AssignmentExpression" || node.parent.type === "AssignmentPattern") {
                    if (node.parent.right.type === "Identifier") {
                        objectName = node.parent.right.name;
                    }
                }
```

Reply:
> why the CI was failing

It's failing on other PRs too. Seems unrelated to any changes we made, so it's likely a problem in dependencies, I don't think anything needs to be changed here, we'll just re-run the checks when the problem gets fixed.

Reply:
Browser test is working now.